libs/agno/agno/workflow/v2/workflow.py

```diff
@@ -127,6 +149,50 @@ def initialize_workflow(self):
 
         self._update_workflow_session_state()
 
+    def _handle_event(self, event: "WorkflowRunResponseEvent", workflow_run_response: WorkflowRunResponse) -> None:
+        """Handle workflow events for storage - similar to Team._handle_event"""
+        if not self.store_events:
+            return
+
+        # Check if this event type should be skipped
+        if self.events_to_skip:
+            event_type = event.event
+            for skip_event in self.events_to_skip:
+                if isinstance(skip_event, str):
+                    if event_type == skip_event:
+                        return
+                else:
+                    # It's a WorkflowRunEvent enum
+                    if event_type == skip_event.value:
+                        return
+
+        # Store the event
+        if workflow_run_response.events is None:
+            workflow_run_response.events = []
+
+        workflow_run_response.events.append(event)
+
+    def _transform_step_output_to_event(
+        self, step_output: StepOutput, workflow_run_response: WorkflowRunResponse, step_index: Optional[int] = None
+    ) -> StepOutputEvent:
+        """Transform a StepOutput object into a StepOutputEvent for consistent streaming interface"""
+        return StepOutputEvent(
+            run_id=workflow_run_response.run_id or "",
+            workflow_name=workflow_run_response.workflow_name,
+            workflow_id=workflow_run_response.workflow_id,
+            session_id=workflow_run_response.session_id,
+            step_name=step_output.step_name,
+            step_index=step_index,
+            content=step_output.content,
+            images=step_output.images,
+            videos=step_output.videos,
+            audio=step_output.audio,
+            success=step_output.success,
+            error=step_output.error,
+            stop=step_output.stop,
+            step_output=step_output,
```

Comment:
This feels very duplicated.... could we not just have whatever is on the inside on the root?

Reply:
makes sense, updated!